src/Definitions/CallableDefinition.php

```diff
@@ -19,8 +19,8 @@ public function __construct(callable $method)
     public function resolve(ContainerInterface $container)
     {
         $callback = $this->method;
-        if ($container->has(Injector::class)) {
-            return $container->get(Injector::class)->invoke($callback);
+        if (class_exists(Injector::class)) {
+            return (new Injector($container))->invoke($callback);
```

Comment:
Should we optimize creation of new injectors every time?
Something like this:

```suggestion
            return $this->getInjector($container)->invoke($callback);
```

```php
    private static array $injectors = [];
    private function getInjector(ContainerInterface $container): Injector
    {
        $id = spl_object_hash($container);
        if (self::$injectors[$id] === null) {
            self::$injectors[$id] = new Injector($container);
        }
        return self::$injectors[$id];
    }
```

On the other side:

- `Injector` is very simple object and creation is cheap
- `spl_object_hash` has a flaw (but it is solvable)

Reply:
Done